### PR TITLE
Log Pose2D & Rotation

### PR DESCRIPTION
**Pull request by Asher123456789**
We added Pose2D and its rotation to SmartDashboard.

### CODE REVIEW

**Review by azvanderpas — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Telemetry.java [hunk 1]
```diff
@@ -25,11 +26,14 @@ public class Telemetry {
    */
   public Telemetry(double maxSpeed) {
     MaxSpeed = maxSpeed;
```
Comment (lohengrin332) [inline src/main/java/frc/robot/Telemetry.java:28]: I know this is pre-exiating code, but idiomatic Java usually uses `lowerCaseFirst` naming for variables, and `UpperCaseFirst` for classes. Please update this variable accordingly, and then you can differentiate it from the parameter of the same name using `this.maxSpeed`. E.g.:

```suggestion
    this.maxSpeed = maxSpeed;
```

File: src/main/java/frc/robot/Telemetry.java [hunk 2]
```diff
@@ -25,11 +26,14 @@ public class Telemetry {
...
+    SmartDashboard.putData(field);
   }
 
   /* What to publish over networktables for telemetry */
   NetworkTableInstance inst = NetworkTableInstance.getDefault();
 
+  Field2d field = new Field2d();
```
Comment (lohengrin332) [inline src/main/java/frc/robot/Telemetry.java:35]: Please use `final` for variables whenever possible. See [this Q&A from StackOverflow](https://stackoverflow.com/questions/27369584/what-does-the-jvm-do-with-a-final-variable-in-a-class) for one reason why.

```suggestion
  final Field2d field = new Field2d();
```

File: src/main/java/frc/robot/Telemetry.java [hunk 3]
```diff
@@ -25,11 +26,14 @@ public class Telemetry {
...
+
   /* Robot pose for field positioning */
   NetworkTable table = inst.getTable("Pose");
   DoubleArrayPublisher fieldPub = table.getDoubleArrayTopic("robotPose").publish();
```
[the remark this answers is not in the document]
Comment (ModBoyex) [reply src/main/java/frc/robot/Telemetry.java:39]: but it doesn not allow you to drive the robot in simulation. Maybe there a better way to do it, but what Asher did allowed you to drive it.

File: src/main/java/frc/robot/Telemetry.java (hunk 1 above)
Comment (ModBoyex) [reply src/main/java/frc/robot/Telemetry.java:28]: This gives me an error when changing it to this. I leave that for another pull request.

File: src/main/java/frc/robot/Telemetry.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (ModBoyex) [reply src/main/java/frc/robot/Telemetry.java:39]: Never mind, found were it logs it in networktables. Asher's code is now obsolete.

**Pull request closed without merging**